node-test/src/test/scala/org/bitcoins/node/networking/peer/DataMessageHandlerTest.scala

```diff
@@ -143,10 +153,13 @@ class DataMessageHandlerTest extends NodeUnitTest {
         payload =
           CompactFilterMessage(FilterType.Basic, hash.flip, filter.filter.bytes)
 
-        callbacks = NodeCallbacks.onCompactFilterReceived(callback)
-        _ = nodeConfig.addCallbacks(callbacks)
+        nodeCallbacks = NodeCallbacks.onCompactFilterReceived(callback)
+        _ = spv.nodeAppConfig.addCallbacks(nodeCallbacks)
+        dataMessageHandler =
+          DataMessageHandler(genesisChainApi)(spv.executionContext,
+                                              spv.nodeAppConfig,
+                                              spv.chainConfig)
 
-        dataMessageHandler = DataMessageHandler(genesisChainApi)
         _ <- dataMessageHandler.handleDataPayload(payload, sender)
```

Comment:
I don't think the changes on this file are needed, #2476 should have fixed this

Reply:
ive rebased ontop of this so i guess it is necessary